tests/others/test_metrics.py

```diff
@@ -74,7 +74,7 @@ def test_connections_manager_integration(self):
         self.assertEquals(manager.metrics.known_peers, 3)
         self.assertEquals(manager.metrics.connected_peers, 2)
         self.assertEquals(manager.metrics.handshaking_peers, 1)
-        self.assertEquals(manager.metrics.connecting_peers, 0)
+        self.assertEquals(manager.metrics.connecting_peers, 1)
```

Comment:
Because before we were using a `Clock` as a reactor for tests, and `Clock` does not implement the `IReactorTCP` interface, which it should since this interface's methods are actually used by us. For that, when trying to make the connection, the reactor would raise when exception when the non-existent method was called.

By changing our reactor tests to the `MemoryReactorClock`, which [does implement](https://github.com/twisted/twisted/blob/39ee213fca02e08bc1aadad61607d7e7e123a776/src/twisted/internet/testing.py#L407-L409) `IReactorTCP`, the connection we create correctly gets to the "connecting" state, and that's why the test fixture is changed (it never get's to "connected", though).

Reply:
Got it. Why wasn't the test failing when the non-existent method was called? Anyway, I agree with your change and you can mark it as resolved.

Reply:
Because there's an error callback on the deferred that tries to make the connection: https://github.com/HathorNetwork/hathor-core/blob/master/hathor/p2p/manager.py#L570-L574

This callback simply emits a warning log and a PubSub event, and removes the connection from `self.connecting_peers`. That's why it was `0` in the old code. In the new code this error callback is not called, so the connection stays in `self.connecting_peers`, keeping it equal to `1`.

Reply:
Then it was a bug in the test. Thanks!